### PR TITLE
[GCS] Use new interface class GcsClient in ray

### PR DESCRIPTION
**Pull request by micafan**
## Why are these changes needed?

Use GcsClient instead of RedisGcsClient in ray.

## Related issue number
Closes #5058

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Review by zhijunfu — APPROVED** [review]
LGTM, thanks!

File: src/ray/gcs/redis_gcs_client.h [hunk 1]
```diff
@@ -77,41 +62,38 @@ class RAY_EXPORT RedisGcsClient : public GcsClient {
...
-  std::string DebugString() const;
-
- private:
-  /// Attach this client to an asio event loop. Note that only
-  /// one event loop should be attached at a time.
-  void Attach(boost::asio::io_service &io_service);
-
-  /// The following three methods will be deprecated, use method Actors() instead.
+  /// The following xxx_table methods is no longer used as interface,
+  /// just used as implementation.
```
Comment (stephanie-wang) [inline src/ray/gcs/redis_gcs_client.h]:
```suggestion
  /// The following xxx_table methods implement the Accessor interfaces.
```

File: src/ray/gcs/redis_gcs_client.h [hunk 2]
```diff
@@ -77,41 +62,38 @@ class RAY_EXPORT RedisGcsClient : public GcsClient {
...
+  /// Use new interface method Actors() instead of the following three methods.
```
Comment (stephanie-wang) [inline src/ray/gcs/redis_gcs_client.h]:
```suggestion
  /// Implements the Actors() interface.
```

Same for below.

**Review by stephanie-wang — APPROVED** [review]

File: src/ray/gcs/redis_gcs_client.h (hunk 1 above)
Comment (micafan) [reply src/ray/gcs/redis_gcs_client.h]: done

File: src/ray/gcs/redis_gcs_client.h (hunk 2 above)
Comment (micafan) [reply src/ray/gcs/redis_gcs_client.h]: done

**Pull request merged**